docs(span-first): Add Noop Span specification to Span API (#16524)
getsentry/sentry-docs: develop-docs/sdk/telemetry/spans/span-api.mdx

At this time, SDKs MUST NOT expose APIs like `Span::startChild` or similar functionality that explicitly creates a child span.
This is still TBD but the `parentSpan` option should suffice to serve this use case.

## Noop Span

A Noop span is a span that implements the full [Span Interface](#span-interface) but does not record any data and has no influence on the trace hierarchy.

> `startSpan` **MUST** always return a valid `Span` instance. When a span should not be recorded (e.g., due to [sampling](../sampling/) or [filtering](../filtering/) decisions), the SDK **MUST** return a Noop span. This allows callers to use the span without null checks.

Reviewer: ```suggestion
`startSpan` **MUST** always return a valid `Span` instance. If a span should not be recorded (for example due to [sampling](../sampling/) or [filtering](../filtering/) decisions), the SDK **MUST** return a Noop span instead. This allows callers to use the span without null checks.
```
